### PR TITLE
fix: Keep Docker Sidecar Alive for Data Retrieval

### PR DESCRIPTION
**Pull request by ishaansehgal99**
**Reason for Change**:
- Keep the docker sidecar container alive so incase ACR push fails we can still exec into the container to retrieve completed tuning job files in the /mnt/results folder.

- Perform validation check for output images that are uppercase

- Use projected volume for mounting docker config secrets - mount at directory location /root/.docker/config

- Add secret for E2E Temp ACR

### CODE REVIEW

File: api/v1alpha1/workspace_validation.go [hunk 1]
```diff
@@ -271,7 +271,19 @@ func (r *DataDestination) validateCreate() (errs *apis.FieldError) {
 		re := regexp.MustCompile(`^(.+/[^:/]+):([^:/]+)$`)
 		if !re.MatchString(r.Image) {
 			errs = errs.Also(apis.ErrInvalidValue("Invalid image format, require full output image URL", "Image"))
+		} else {	// Executes if image is of correct format
```
Comment (Fei-Guo) [inline api/v1alpha1/workspace_validation.go:274]: Should you do the same check for the data source as well? or maybe have a common function due to the duplicated code.

Comment (Fei-Guo) [reply api/v1alpha1/workspace_validation.go:274]: And add UT pls

File: test/e2e/preset_test.go [hunk 2]
```diff
@@ -56,6 +56,7 @@ func loadTestEnvVars() {
 	aiModelsRegistry = utils.GetEnv("AI_MODELS_REGISTRY")
 	aiModelsRegistrySecret = utils.GetEnv("AI_MODELS_REGISTRY_SECRET")
 	supportedModelsYamlPath = utils.GetEnv("SUPPORTED_MODELS_YAML_PATH")
+	azureClusterName = utils.GetEnv("AZURE_CLUSTER_NAME")
```
Comment (Fei-Guo) [inline test/e2e/preset_test.go:59]: Is this used anywhere?

Comment (ishaansehgal99) [reply test/e2e/preset_test.go:59]: Yes this is used by fine tuning e2e - outputRegistryUrl. e2e for fine tuning previously was uploading to local ACR now we are using temp ACR

File: api/v1alpha1/workspace_validation.go (hunk 1 above)
Comment (ishaansehgal99) [reply api/v1alpha1/workspace_validation.go]: cool added function and added UTs

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**